tiledb/cf/creator.py

```diff
@@ -698,14 +717,22 @@ def __init__(
         sparse: bool = False,
     ):
         self._registry = ArrayRegistry(dataspace_registry, name, dims)
+        self._domain_creator = DomainCreator(self._registry)
         self.cell_order = cell_order
         self.tile_order = tile_order
         self.capacity = capacity
         if tiles is not None:
-            self.tiles = tiles
+            if len(tiles) != self.ndim:
+                raise ValueError(
+                    f"Cannot set tiles. Got {len(tiles)} tile(s) for an array with "
+                    f"{self.ndim} dimension(s)."
+                )
+            for dim_creator, tile in zip(self._domain_creator, tiles):
+                dim_creator.tile = tile
         self.coords_filters = coords_filters
         if dim_filters is not None:
-            self.dim_filters = dim_filters
+            for dim_name, filters in dim_filters.items():
+                self.domain_creator.dim_creator(dim_name).filters = filters
```

Comment:
Should it be self._domain_creator instead of self.domain_creator as above in setting tiles directly from attribute _domain_creator.